File: 645.set-mismatch/solution.scala
import scala.collection.mutable.BitSet

object Solution {
    def findErrorNums(nums: Array[Int]): Array[Int] = {
        val bs = new BitSet(nums.length + 1)
        var dp = 0
        for (n <- nums)
            if (!bs.add(n)) dp = n
        for (i <- 1 to nums.length)
            if (!bs.contains(i))
                return Array(dp, i)
        Array(dp, 0)
    }
}

File: 645.set-mismatch/solution.go
func findErrorNums(nums []int) []int {
    h := make([]int, len(nums) + 1)
    for _, n := range nums {
        h[n]++
    }
    res := make([]int, 2)
    for i := 1; i <= len(nums); i++ {
        switch h[i] {
            case 0: res[1] = i
            case 2: res[0] = i
        } 
    }
    return res
}